src/ParamPassing.cc

```diff
@@ -24,45 +24,88 @@ namespace sdf
 inline namespace SDF_VERSION_NAMESPACE {
 
 //////////////////////////////////////////////////
-void updateParams(const tinyxml2::XMLElement *_childXmlParams,
+void updateParams(tinyxml2::XMLElement *_childXmlParams,
                   SDFPtr _includeSDF, Errors &_errors)
 {
   // loop through <experimental:params> children
-  const tinyxml2::XMLElement *childElemXml = nullptr;
+  tinyxml2::XMLElement *childElemXml = nullptr;
   for (childElemXml = _childXmlParams->FirstChildElement();
        childElemXml;
        childElemXml = childElemXml->NextSiblingElement())
   {
     // element identifier
-    const char *childElemId = childElemXml->Attribute("name");
+    const char *childElemId = childElemXml->Attribute("element_id");
     if (!childElemId)
     {
       tinyxml2::XMLPrinter printer;
       childElemXml->Accept(&printer);
 
       _errors.push_back({ErrorCode::ATTRIBUTE_MISSING,
-        "Element identifier requires a name, but the name is not set. "
-        "Skipping element modification:\n"
+        "Element identifier requires an element_id attribute, but the "
+        "element_id is not set. Skipping element modification:\n"
         + std::string(printer.CStr())
       });
       continue;
     }
 
+    // *** Retrieve specified element using element identifier ***
+
     std::string actionStr;
     if (childElemXml->Attribute("action"))
       actionStr = std::string(childElemXml->Attribute("action"));
 
     // get element pointer to specified element using element identifier
-    ElementPtr  elem;
+    ElementPtr elem = getElementById(_includeSDF, childElemXml->Name(),
+                          childElemId);
+
+    std::string elemIdAttr(childElemId);
+
     if (actionStr == "add")
     {
-      // TODO(jenn) implement this, skipping for now
-      continue;
-    }
-    else
-    {
-      elem = getElementById(_includeSDF, childElemXml->Name(),
-                            childElemId, true);
+      if (elem != nullptr)
+      {
+        // TODO(jenn) make test for this
+        tinyxml2::XMLPrinter printer;
+        childElemXml->Accept(&printer);
+
+        _errors.push_back({ErrorCode::DUPLICATE_NAME,
+          "Could not add element <" + std::string(childElemXml->Name())
+          + " element_id='" + childElemXml->Attribute("element_id")
+          + "'> because element already exists in included model. "
+          + "Skipping element addition:\n"
+          + printer.CStr()
+        });
+        continue;
+      }
+
+      size_t found = elemIdAttr.rfind("::");
+      if (found != std::string::npos)
+      {
+        // +2 past double colons
+        elemIdAttr = elemIdAttr.substr(found+2);
+
+        // TODO(jenn) add test for this (e.g., elemChildId='test::')
+        if (elemIdAttr.empty())
+        {
+          std::cout << "ADD error" << std::endl;
```

Comment:
Is this suppose to be replaced with an error message?

Reply:
Yes, I've updated this to an error message, which happens when the `element_id` is missing a name after `::` (e.g., `<link element_id="chassis::"/>`): 5f648cc55506a3604237e533a7385cd43d8aa34f